client/control.go

```diff
@@ -134,7 +140,12 @@ func (ctl *Control) HandleReqWorkConn(inMsg *msg.ReqWorkConn) {
 	}
 
 	m := &msg.NewWorkConn{
-		RunId: ctl.runId,
+		RunId:     ctl.runId,
+		Timestamp: time.Now().Unix(),
```

Comment:
(non-voting)

What is the purpose on having to handle `Timestamp` in this object and others? Why can't `GetAuthKey()` handle that itself internally? 

If the answer is that it can, then we should create another PR to modify that and simplify the code all over.

Reply:
+@fatedier: I'm kind of in favour of keeping Timestamp on any case. It's an informative time that's added to this request, unrelated to the fact it helps with auth.